app/src/org/commcare/activities/PostRequestActivity.java

```diff
@@ -108,13 +112,8 @@ private void performSync() {
 
     private void makePostRequest() {
         if (!hasTaskLaunched && !inErrorState) {
-            ModernHttpTask postTask;
-            try {
-                postTask = new ModernHttpTask(this, url, params, true, null);
-            } catch (ModernHttpRequester.PlainTextPasswordException e) {
-                enterErrorState(Localization.get("post.not.using.https", url.toString()));
-                return;
-            }
+            RequestBody requestBody = ModernHttpRequester.getPostBody(params);
+            ModernHttpTask postTask = new ModernHttpTask(this, url.toString(), new HashMap(), requestBody, HTTPMethod.POST, null);
```

Comment:
I don't think that it should be the responsibility of `ModernHttpRequester` to create the Request body as in general different  POST requests would always have different request bodies. Though to make it a bit cleaner we can make an overloaded constructor for `ModernHttpRequester`